[2024-11-21T10:10:28.971+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-11-21T10:10:28.987+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: etl_pipeline.extract_google_trends manual__2024-11-21T10:10:26.259501+00:00 [queued]>
[2024-11-21T10:10:28.996+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: etl_pipeline.extract_google_trends manual__2024-11-21T10:10:26.259501+00:00 [queued]>
[2024-11-21T10:10:28.997+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 1
[2024-11-21T10:10:29.012+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): extract_google_trends> on 2024-11-21 10:10:26.259501+00:00
[2024-11-21T10:10:29.019+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=2677) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-11-21T10:10:29.021+0000] {standard_task_runner.py:72} INFO - Started process 2679 to run task
[2024-11-21T10:10:29.021+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'etl_pipeline', 'extract_google_trends', 'manual__2024-11-21T10:10:26.259501+00:00', '--job-id', '200', '--raw', '--subdir', 'DAGS_FOLDER/dag.py', '--cfg-path', '/tmp/tmpl0zc6u6_']
[2024-11-21T10:10:29.022+0000] {standard_task_runner.py:105} INFO - Job 200: Subtask extract_google_trends
[2024-11-21T10:10:29.065+0000] {task_command.py:467} INFO - Running <TaskInstance: etl_pipeline.extract_google_trends manual__2024-11-21T10:10:26.259501+00:00 [running]> on host 7a40aa914b11
[2024-11-21T10:10:29.363+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='etl_pipeline' AIRFLOW_CTX_TASK_ID='extract_google_trends' AIRFLOW_CTX_EXECUTION_DATE='2024-11-21T10:10:26.259501+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-11-21T10:10:26.259501+00:00'
[2024-11-21T10:10:29.365+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2024-11-21T10:10:29.365+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2024-11-21T10:10:29.366+0000] {logging_mixin.py:190} INFO - Current task name:extract_google_trends state:running start_date:2024-11-21 10:10:28.988923+00:00
[2024-11-21T10:10:29.367+0000] {logging_mixin.py:190} INFO - Dag name:etl_pipeline and current dag run status:running
[2024-11-21T10:10:29.368+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-11-21T10:10:29.372+0000] {extract_google_trends.py:124} INFO - Starting Google Trends data scraping...
[2024-11-21T10:10:29.374+0000] {extract_google_trends.py:57} INFO - Running Apify actor for 'sunscreen skin aqua'...
[2024-11-21T10:10:31.199+0000] {extract_google_trends.py:17} INFO - Sent request to Apify API to run actor with status code: 201
[2024-11-21T10:10:32.270+0000] {extract_google_trends.py:31} INFO - Checked actor run status with status code: 200
[2024-11-21T10:10:32.273+0000] {extract_google_trends.py:89} INFO - Actor is still running for 'sunscreen skin aqua'... Waiting...
[2024-11-21T10:10:43.451+0000] {extract_google_trends.py:31} INFO - Checked actor run status with status code: 200
[2024-11-21T10:10:43.452+0000] {extract_google_trends.py:89} INFO - Actor is still running for 'sunscreen skin aqua'... Waiting...
[2024-11-21T10:10:54.675+0000] {extract_google_trends.py:31} INFO - Checked actor run status with status code: 200
[2024-11-21T10:10:54.678+0000] {extract_google_trends.py:89} INFO - Actor is still running for 'sunscreen skin aqua'... Waiting...
[2024-11-21T10:11:05.723+0000] {extract_google_trends.py:31} INFO - Checked actor run status with status code: 200
[2024-11-21T10:11:05.726+0000] {extract_google_trends.py:89} INFO - Actor is still running for 'sunscreen skin aqua'... Waiting...
[2024-11-21T10:11:16.773+0000] {extract_google_trends.py:31} INFO - Checked actor run status with status code: 200
[2024-11-21T10:11:16.775+0000] {extract_google_trends.py:89} INFO - Actor is still running for 'sunscreen skin aqua'... Waiting...
[2024-11-21T10:11:27.949+0000] {extract_google_trends.py:31} INFO - Checked actor run status with status code: 200
[2024-11-21T10:11:27.956+0000] {extract_google_trends.py:89} INFO - Actor is still running for 'sunscreen skin aqua'... Waiting...
[2024-11-21T10:11:39.146+0000] {extract_google_trends.py:31} INFO - Checked actor run status with status code: 200
[2024-11-21T10:11:39.151+0000] {extract_google_trends.py:89} INFO - Actor is still running for 'sunscreen skin aqua'... Waiting...
[2024-11-21T10:11:50.570+0000] {extract_google_trends.py:31} INFO - Checked actor run status with status code: 200
[2024-11-21T10:11:50.572+0000] {extract_google_trends.py:89} INFO - Actor is still running for 'sunscreen skin aqua'... Waiting...
[2024-11-21T10:12:01.780+0000] {extract_google_trends.py:31} INFO - Checked actor run status with status code: 200
[2024-11-21T10:12:01.782+0000] {extract_google_trends.py:89} INFO - Actor is still running for 'sunscreen skin aqua'... Waiting...
[2024-11-21T10:12:12.957+0000] {extract_google_trends.py:31} INFO - Checked actor run status with status code: 200
[2024-11-21T10:12:12.959+0000] {extract_google_trends.py:89} INFO - Actor is still running for 'sunscreen skin aqua'... Waiting...
[2024-11-21T10:12:24.059+0000] {extract_google_trends.py:31} INFO - Checked actor run status with status code: 200
[2024-11-21T10:12:24.062+0000] {extract_google_trends.py:89} INFO - Actor is still running for 'sunscreen skin aqua'... Waiting...
[2024-11-21T10:12:35.146+0000] {extract_google_trends.py:31} INFO - Checked actor run status with status code: 200
[2024-11-21T10:12:35.150+0000] {extract_google_trends.py:89} INFO - Actor is still running for 'sunscreen skin aqua'... Waiting...
[2024-11-21T10:12:46.330+0000] {extract_google_trends.py:31} INFO - Checked actor run status with status code: 200
[2024-11-21T10:12:46.333+0000] {extract_google_trends.py:89} INFO - Actor is still running for 'sunscreen skin aqua'... Waiting...
[2024-11-21T10:12:57.445+0000] {extract_google_trends.py:31} INFO - Checked actor run status with status code: 200
[2024-11-21T10:12:57.447+0000] {extract_google_trends.py:87} INFO - Actor run succeeded for 'sunscreen skin aqua'.
[2024-11-21T10:12:59.171+0000] {extract_google_trends.py:45} INFO - Fetched dataset with status code: 200
[2024-11-21T10:12:59.282+0000] {extract_google_trends.py:105} INFO - Data for 'sunscreen skin aqua' saved to 'dags/hasil data extract/GoogleTrend_sunscreen_skin_aqua.json'.
[2024-11-21T10:12:59.333+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/etl/extract_google_trends.py", line 125, in extract_google_trends
    scrape_and_process_apify(api_token=api_token, actor_id=actor_id, search_terms=search_terms, output_folder=output_folder)
  File "/opt/airflow/dags/etl/extract_google_trends.py", line 112, in scrape_and_process_apify
    writer.writerows(dataset)
  File "/usr/local/lib/python3.12/csv.py", line 167, in writerows
    return self.writer.writerows(map(self._dict_to_list, rowdicts))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/csv.py", line 159, in _dict_to_list
    raise ValueError("dict contains fields not in fieldnames: "
ValueError: dict contains fields not in fieldnames: 'maxItems', 'maxConcurrency', 'isPublic', 'skipDebugScreen', 'geo', 'isMultiple', 'maxRequestRetries', 'pageLoadTimeoutSecs', 'viewedFrom', 'timeRange', 'searchTerms'
[2024-11-21T10:12:59.507+0000] {logging_mixin.py:190} INFO - Task instance in failure state
[2024-11-21T10:12:59.509+0000] {logging_mixin.py:190} INFO - Task start:2024-11-21 10:10:28.988923+00:00 end:2024-11-21 10:12:59.490688+00:00 duration:150.501765
[2024-11-21T10:12:59.512+0000] {logging_mixin.py:190} INFO - Task:<Task(PythonOperator): extract_google_trends> dag:<DAG: etl_pipeline> dagrun:<DagRun etl_pipeline @ 2024-11-21 10:10:26.259501+00:00: manual__2024-11-21T10:10:26.259501+00:00, state:running, queued_at: 2024-11-21 10:10:26.573591+00:00. externally triggered: True>
[2024-11-21T10:12:59.513+0000] {logging_mixin.py:190} INFO - Failure caused by dict contains fields not in fieldnames: 'maxItems', 'maxConcurrency', 'isPublic', 'skipDebugScreen', 'geo', 'isMultiple', 'maxRequestRetries', 'pageLoadTimeoutSecs', 'viewedFrom', 'timeRange', 'searchTerms'
[2024-11-21T10:12:59.515+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=etl_pipeline, task_id=extract_google_trends, run_id=manual__2024-11-21T10:10:26.259501+00:00, execution_date=20241121T101026, start_date=20241121T101028, end_date=20241121T101259
[2024-11-21T10:12:59.571+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-11-21T10:12:59.575+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 200 for task extract_google_trends (dict contains fields not in fieldnames: 'maxItems', 'maxConcurrency', 'isPublic', 'skipDebugScreen', 'geo', 'isMultiple', 'maxRequestRetries', 'pageLoadTimeoutSecs', 'viewedFrom', 'timeRange', 'searchTerms'; 2679)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/etl/extract_google_trends.py", line 125, in extract_google_trends
    scrape_and_process_apify(api_token=api_token, actor_id=actor_id, search_terms=search_terms, output_folder=output_folder)
  File "/opt/airflow/dags/etl/extract_google_trends.py", line 112, in scrape_and_process_apify
    writer.writerows(dataset)
  File "/usr/local/lib/python3.12/csv.py", line 167, in writerows
    return self.writer.writerows(map(self._dict_to_list, rowdicts))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/csv.py", line 159, in _dict_to_list
    raise ValueError("dict contains fields not in fieldnames: "
ValueError: dict contains fields not in fieldnames: 'maxItems', 'maxConcurrency', 'isPublic', 'skipDebugScreen', 'geo', 'isMultiple', 'maxRequestRetries', 'pageLoadTimeoutSecs', 'viewedFrom', 'timeRange', 'searchTerms'
[2024-11-21T10:12:59.609+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-11-21T10:12:59.687+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-11-21T10:12:59.696+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
